cli/src/main/scala/com/codacy/analysis/cli/Main.scala

```diff
@@ -84,6 +91,22 @@ class MainImpl extends CLIApp {
     }).getOrElse(Right(()))
   }
 
+  private def validateCommitUuid(projectDirectory: File, commitUuidOpt: Option[Commit.Uuid]): Either[CLIError, Unit] = {
+    (for {
+      repo <- Git.repository(projectDirectory).toOption
+      gitCommit <- repo.latestCommit.toOption
+      paramCommitUuid <- commitUuidOpt
+    } yield {
+      if (gitCommit.commitUuid != paramCommitUuid) {
+        val error = CLIError.CommitUuidsDoNotMatch(paramCommitUuid, gitCommit.commitUuid)
+        logger.error(error.message)
+        Left(error)
+      } else {
+        Right(())
+      }
+    }).getOrElse(Right(()))
+  }
```

Comment:
The "if" can be moved into the for comprehension so we have a single Right case. Check if you think it's cleaner or harder to understand that way.